plugin/documents.py

```diff
@@ -639,13 +639,9 @@ def render_highlights_on_main_thread() -> None:
     # --- textDocument/complete ----------------------------------------------------------------------------------------
 
     def _on_query_completions_async(self, resolve_completion_list: ResolveCompletionsFn, location: int) -> None:
-        sessions = self.sessions('completionProvider')
```

Comment:
`sessions = tuple(self.sessions('completionProvider'))` may also work.

Reply:
Why `tuple` and not `list`?

BTW. that sounds like a better solution because we wouldn't call `self.purge_changes_async()` unnecessarily when there are no sessions.
And you would only need this one-line change.

Reply:
Yeah I think this can be a one-liner

Reply:
`tuple` because there is no need to manipulate the set of sessions.